lucene/sandbox/src/java/org/apache/lucene/sandbox/rabitq/SpaceUtils.java

```diff
@@ -98,7 +115,12 @@ public static long[] transposeBin(byte[] q, int D, int B_QUERY) {
       qOffset += 32;
     }
 
-    return quantQuery;
+    ByteBuffer bb = ByteBuffer.allocate((B / 8) * B_QUERY);
```

Comment:
If we absolutely need to still know about the `4`, then we should make it a public static final int somewhere and refer to it directly.